packages/investor-foxy/src/api/api.ts

```diff
@@ -0,0 +1,856 @@
+  }
+
+  async removeLiquidity(input: TxInput): Promise<string> {
+    const {
+      amountDesired,
+      accountNumber = 0,
+      dryRun = false,
+      contractAddress,
+      userAddress,
+      wallet
+    } = input
+    this.verifyAddresses([userAddress, contractAddress])
+    if (!amountDesired.gt(0)) throw new Error('Must send valid amount')
+    if (!wallet) throw new Error('Missing inputs')
+
+    let estimatedGasBN: BigNumber
+    try {
+      estimatedGasBN = await this.estimateRemoveLiquidityGas(input)
+    } catch (e) {
+      throw new Error(`Estimate Gas Error: ${e}`)
+    }
+
+    const liquidityReserveContract = this.liquidityReserveContracts.find(
+      (item) => toLower(item.options.address) === toLower(contractAddress)
+    )
+    if (!liquidityReserveContract) throw new Error('Not a valid contract address')
+
+    const data: string = liquidityReserveContract.methods
+      .removeLiquidity(amountDesired.toString())
+      .encodeABI({
+        from: userAddress
+      })
+    const nonce = await this.web3.eth.getTransactionCount(userAddress)
+    const gasPrice = await this.web3.eth.getGasPrice()
+    const estimatedGas = estimatedGasBN.toString()
+    const bip44Params = this.adapter.buildBIP44Params({ accountNumber })
+    const chainId = Number(this.network)
+    const payload = {
+      bip44Params,
+      chainId,
+      data,
+      estimatedGas,
+      gasPrice,
+      nonce: String(nonce),
+      to: contractAddress,
+      value: '0'
+    }
+    const txToSign = buildTxToSign(payload)
+    if (wallet.supportsOfflineSigning()) {
+      const signedTx = await this.adapter.signTransaction({ txToSign, wallet })
+      if (dryRun) return signedTx
+      return this.broadcastTx(signedTx)
+    } else if (wallet.supportsBroadcast() && this.adapter.signAndBroadcastTransaction) {
+      if (dryRun) {
+        throw new Error(`Cannot perform a dry run with wallet of type ${wallet.getVendor()}`)
+      }
+      return this.adapter.signAndBroadcastTransaction({ txToSign, wallet })
+    } else {
+      throw new Error('Invalid HDWallet configuration ')
+    }
+  }
+
+  // returns time in seconds until withdraw request is claimable
+  // dependent on rebases happening when epoch.expiry block is reached
+  async getTimeUntilClaimable(input: TxInputWithoutAmountAndWallet): Promise<string> {
+    const { contractAddress, userAddress } = input
+    this.verifyAddresses([userAddress, contractAddress])
+
+    const stakingContract = this.foxyStakingContracts.find(
+      (item) => toLower(item.options.address) === toLower(contractAddress)
+    )
+    if (!stakingContract) throw new Error('Not a valid contract address')
+
+    const coolDownInfo = await stakingContract.methods.coolDownInfo(userAddress).call()
+    const epoch = await stakingContract.methods.epoch().call()
+    const currentBlock = await this.web3.eth.getBlockNumber()
+    const epochsLeft = coolDownInfo.expiry - epoch.number - 1 // epochs left after the current one
```

Comment:
we might want to alias this to a differently named variable better describing what it does

Reply:
also to be defensive check if this is ever negative